test/e2e/framework/skipper/skipper.go

```diff
@@ -137,20 +129,26 @@ func SkipUnlessAtLeast(value int, minValue int, message string) {
 
 // SkipUnlessLocalEphemeralStorageEnabled skips if the LocalStorageCapacityIsolation is not enabled.
 func SkipUnlessLocalEphemeralStorageEnabled() {
-	if !utilfeature.DefaultFeatureGate.Enabled(localStorageCapacityIsolation) {
-		skipInternalf(1, "Only supported when %v feature is enabled", localStorageCapacityIsolation)
+	if !utilfeature.DefaultFeatureGate.Enabled(features.LocalStorageCapacityIsolation) {
+		skipInternalf(1, "Only supported when %v feature is enabled", features.LocalStorageCapacityIsolation)
 	}
 }
 
 func SkipUnlessDownwardAPIHugePagesEnabled() {
-	if !utilfeature.DefaultFeatureGate.Enabled(downwardAPIHugePages) {
-		skipInternalf(1, "Only supported when %v feature is enabled", downwardAPIHugePages)
+	if !utilfeature.DefaultFeatureGate.Enabled(features.DownwardAPIHugePages) {
+		skipInternalf(1, "Only supported when %v feature is enabled", features.DownwardAPIHugePages)
 	}
 }
 
 func SkipUnlessExecProbeTimeoutEnabled() {
-	if !utilfeature.DefaultFeatureGate.Enabled(execProbeTimeout) {
-		skipInternalf(1, "Only supported when %v feature is enabled", execProbeTimeout)
+	if !utilfeature.DefaultFeatureGate.Enabled(features.ExecProbeTimeout) {
+		skipInternalf(1, "Only supported when %v feature is enabled", features.ExecProbeTimeout)
+	}
+}
+
+func SkipUnlessNetworkPolicyEndPortEnabled() {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.NetworkPolicyEndPort) {
```

Comment:
I was checking this again, and I think that this doesn't work as I supposed, I think that this only works internally in the different components , and is useless for e2e test.
The e2e test should use the tag [Feature:...] for filtering sing the ginkgo regex.
@BenTheElder can you confirm?

Reply:
>  The e2e test should use the tag [Feature:...] for filtering sing the ginkgo regex.

@JornShen you will need to change this in the Netpol jobs 
https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-network/sig-network-misc.yaml#L54
https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-network/sig-network-misc.yaml#L603

Reply:
@aojea I found the 'right way' maybe of doing this. Because the e2e test does not create the featureflag map the same way the e2e_node does here (https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/services/services.go#L116) my suggestion here is to make:

```
func SkipUnlessNetworkPolicyEndPortEnabled() {
	if enabled, ok := framework.TestContext.FeatureGates[string(features.NetworkPolicyEndPort)]; !ok || !enabled {
		skipInternalf(1, "Only supported when %v feature is enabled", features.NetworkPolicyEndPort)
	}
}
```

Reply:
first time I see the e2e feature gate option :)
we should check with @spiffxp and @BenTheElder first, it is better if all the e2e tests handles the feature gates the same way

Reply:
@rikatz this gate is opened in e2e.test process. But the one problems is that how to sync the gate status with the testing cluster. If the cluster don't open the gate, e2e.test run this case will still failed.  So this should be config in test-infra repo job config (which open this gate in cluster and open the gate in e2e.test running config) ?

Reply:
I've sent an email to sig-testing, we should get consensus, pick one solution and all reviewers follow that or this will keep happening ...

Reply:
IMO you need to use `[Feature:Foo]` tags.  They allow you to easily create jobs / test suites that only run e2e tests for that feature, and allow you to express your intent to exercise that feature.

Skipping the test if the feature gate is not enabled is great test hygiene, nicely done!  But that by itself doesn't necessarily verify your intent.  Did you _expect_ the feature gate to be enabled when running this test? If so, that should probably be a failure, not a skip.

Reply:
Thanks Aaron, 💯